base/file.jl

```diff
@@ -436,8 +436,13 @@ function readlink(path::AbstractString)
     end
 end
 
-function chmod(p::AbstractString, mode::Integer)
-    err = ccall(:jl_fs_chmod, Int32, (Cstring, Cint), p, mode)
+function chmod(path::AbstractString, mode::Integer; recursive::Bool=false)
+    if recursive && isdir(path)
+        for p in readdir(path)
+            chmod(joinpath(path, p), mode, recursive=true)
+        end
+    end
+    err = ccall(:jl_fs_chmod, Int32, (Cstring, Cint), path, mode)
```

Comment:
I feel like you might want to change the directory's mode first and then recurse, no? That way if you don't have permission, you'll bail out sooner.